compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -855,7 +864,14 @@ class Typer extends Namer
   }
 
   def typedNamedArg(tree: untpd.NamedArg, pt: Type)(using Context): NamedArg = {
-    val arg1 = typed(tree.arg, pt)
+    val arg1 = if (ctx.property(JavaAnnotationArg).isDefined) {
+      pt match {
+        case AppliedType(a, typ :: Nil) if (a.isRef(defn.ArrayClass)) =>
```

Comment:
What about reducing the outer `if/else` by moving the condition to the guard here?